### PR TITLE
Annotate nullability in all private Objective-C only headers

### PR DESCRIPTION
**Pull request by JadenGeller**
I audited our private `.h` files for nullability.

@jpsim

### CODE REVIEW

File: Realm/RLMRealm_Private.h [hunk 1]
```diff
@@ -33,7 +35,7 @@ void RLMRealmTranslateException(NSError **error);
 @interface RLMRealm ()
 
 @property (nonatomic, readonly) BOOL dynamic;
-@property (nonatomic, readwrite) RLMSchema *schema;
+@property (nonatomic, readwrite, null_unspecified) RLMSchema *schema;
```
Comment (jpsim) [inline Realm/RLMRealm_Private.h]: I don't think this can ever be null.

**Comment by jpsim** [comment]
👍

File: Realm/RLMRealm_Private.h (hunk 1 above)
Comment (JadenGeller) [inline Realm/RLMRealm_Private.h]: I was mirroring how it is annotated in the public header.

Comment (jpsim) [inline Realm/RLMRealm_Private.h]: Ah, d80e58b indicates why it has to be `null_unspecified`. 👍

Comment (JadenGeller) [inline Realm/RLMRealm_Private.h]: Looks like it was annotated as such due to a [bug](https://github.com/realm/realm-cocoa/commit/d80e58bd262e) @bdash found. I'll change it if the tests pass.

Comment (JadenGeller) [inline Realm/RLMRealm_Private.h]: Oh, you found it too! (Wish GitHub comments refreshed live 😝)

**Pull request merged**